diff -up links-2.7/configure.in.automake-1_13 links-2.7/configure.in
--- links-2.7/configure.in.automake-1_13	2013-01-05 20:50:43.075381455 +0100
+++ links-2.7/configure.in	2013-01-05 20:51:51.690739865 +0100
@@ -13,7 +13,7 @@ AUTOMAKE="./missing automake"
 AUTOHEADER="./missing autoheader"
 image_formats="GIF PNG XBM"
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
