--- links-2.3/terminal.c.orig	2011-09-28 10:56:51.697735676 +0800
+++ links-2.3/terminal.c	2011-09-28 10:57:32.427742314 +0800
@@ -491,7 +491,7 @@
 
 struct list_head term_specs = {&term_specs, &term_specs};
 
-static struct term_spec dumb_term = { NULL, NULL, "", 0, 1, 0, 0, 0, 0, 0 };
+static struct term_spec dumb_term = { NULL, NULL, "", 1, 1, 0, 0, 1, 0, 0 };
 
 static struct term_spec *get_term_spec(unsigned char *term)
 {
